### PR TITLE
feat: headway widget

### PR DESCRIPTION
**Pull request by rephus**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: https://github.com/lightdash/lightdash-internal-work/issues/264

### Description:
Add headway widget to header 

### Preview:

Peek 2022-04-13 10-01

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by rephus** [comment]
> The bell icon doesn't appear as a clickable button (Like the help icon does). It also appears the Headway widget is attaching to the bottom right of the div block rather than the button. Screenshot 2022-04-12 at 16 38 33

Yeah, no worries, this is not done yet (draft)

File: packages/frontend/src/components/NavBar/HelpMenu/HelpMenu.styles.tsx [hunk 1]
```diff
@@ -72,3 +72,15 @@ export const ItemDescription = styled.p`
...
 `;
+
+export const NotificationWrapper = styled.div`
+    position: relative;
+    color: ${Colors.GRAY5} !important;
+
+    margin-right: 0.625em;
+`;
+export const NotificationWidget = styled.div`
+    position: absolute;
```
Comment (rephus) [inline packages/frontend/src/components/NavBar/HelpMenu/HelpMenu.styles.tsx:83]: https://docs.headwayapp.co/widget#styling-the-badge

File: packages/frontend/src/components/NavBar/HelpMenu/index.tsx [hunk 2]
```diff
@@ -14,15 +14,37 @@ import {
...
     MenuWrapper,
+    NotificationWidget,
+    NotificationWrapper,
 } from './HelpMenu.styles';
 
 const HelpMenu: FC = () => {
+    useEffect(() => {
+        //@ts-ignore
+        if (window.Headway) return; // don't load it twice
+        const script = document.createElement('script');
```
Comment (rephus) [inline packages/frontend/src/components/NavBar/HelpMenu/index.tsx:25]: https://docs.headwayapp.co/widget

Comment (owlas) [reply packages/frontend/src/components/NavBar/HelpMenu/index.tsx:25]: @rephus we have some window-level scripts initialised here: https://github.com/lightdash/lightdash/blob/1d2678418ba6c0edc1de11dfa6372a9b7aacedbe/packages/frontend/src/providers/AppProvider.tsx#L107-L134 - could you add this with those?

Also it looks like the docs say we should run `init` on every component render: https://docs.headwayapp.co/widget

> In Angular/React you should initialize the widget every time the badge element container is re-rendered.

**Review by owlas — CHANGES_REQUESTED** [review]

**Comment by rephus** [comment]
> Also it looks like the docs say we should run init on every component render: https://docs.headwayapp.co/widget

 This is because the script will append a component into the container, that was not needed (actually I had to check if it was already there) . But let me check after moving the script

File: packages/frontend/src/providers/AppProvider.tsx [hunk 3]
```diff
@@ -133,6 +134,16 @@ export const AppProvider: FC = ({ children }) => {
...
 
+    useEffect(() => {
+        if (!isHeadwayLoaded) {
+            const script = document.createElement('script');
+            script.async = false;
+            script.src = 'https://cdn.headwayapp.co/widget.js';
+            document.head.appendChild(script);
+            setIsHeadwayLoaded(true);
+        }
+    }, [isSentryLoaded, setIsSentryLoaded, health]);
```
Comment (owlas) [inline packages/frontend/src/providers/AppProvider.tsx:145]: @rephus dependencies are incorrect

File: packages/frontend/src/providers/AppProvider.tsx [hunk 4]
```diff
@@ -133,6 +134,16 @@ export const AppProvider: FC = ({ children }) => {
...
 
+    useEffect(() => {
+        if (!isHeadwayLoaded) {
+            const script = document.createElement('script');
+            script.async = false;
+            script.src = 'https://cdn.headwayapp.co/widget.js';
+            document.head.appendChild(script);
+            setIsHeadwayLoaded(true);
+        }
+    }, [isHeadwayLoaded, health]);
```
[the remark this answers is not in the document]
Comment (rephus) [reply packages/frontend/src/providers/AppProvider.tsx:145]: mmm, no, it doesn't complain :/

**Pull request merged**